### PR TITLE
tests: new test-integration scripts

### PR DESCRIPTION
**Pull request by davelsan**
## Description

This PR is a re-write of the integration tests command-line interface to only use a few bash scripts to generate the environment and run the tests.

The scripts are written with readability in mind, to facilitate other developers to make changes as the environment requirements evolve, or new features and fixes are requested.

## Changes
- Add a new suite of shell scripts that emulate what a human admin would do to run the tests.
- Add a simple command-line interface that exposes the original options to run the tests.
- Remove unused patch files.
- Add a formatted man page with descriptions and examples of how to use the tests.
- Update the relevant section in the repository `README.md` file.
- Remove the `zendro-env` dependency.
- Bump `package.json` to version 0.5.5.

## Usage

```bash
# Use the already existing command to run the integration tests via npm
npm run test-integration [-- OPTIONS ]

# For more information about the available options, and some examples
npm run test-integration -- -h
```

## Spec

A (somewhat outdated) document outlining the design decisions made by @asishallab can be found in the [gitbook workspace](https://app.gitbook.com/@zendro-dev/s/specifications/zendro-integration-tests) assigned to this PR.

### CODE REVIEW

File: test/testenv_docker_up.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    # Exit with error code 1
+    if [ $elapsedTime == $max_time ]; then
+
+      echo "${RED}${url}${NC} time limit reached"
+      return 1
+
+    fi
+
+    # Wait 2s and rety
```
Comment (coeit) [inline test/testenv_docker_up.sh:28]: rety --> retry

File: test/testenv_generate_code.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  "${GRAPHQL_SERVER_1}/validations/individual.js" \
+  "${TEST_DIR}/integration_test_misc/patches/individual_validate.patch"
+
+patch -V never \
+  "${GRAPHQL_SERVER_1}/validations/accession.js" \
+  "${TEST_DIR}/integration_test_misc/patches/accession_validate_instance1.patch"
+
+patch -V never \
+  "${GRAPHQL_SERVER_1}/models/sql/cat.js" \
+  "${TEST_DIR}/integration_test_misc/patches/cat_readAllCursor.patch"
```
Comment (coeit) [inline test/testenv_generate_code.sh:76]: put in separate `testenv_sync.sh` script

**Review by coeit — CHANGES_REQUESTED** [review]

File: test/testenv_docker_up.sh (hunk 1 above)
Comment (davelsan) [reply test/testenv_docker_up.sh]: Done. I've made the message a bit more descriptive too.

File: test/testenv_generate_code.sh (hunk 2 above)
Comment (davelsan) [reply test/testenv_generate_code.sh]: Done. Patching logic is now in `testenv_sync.sh`.

**Comment by davelsan** [comment]
I'm thinking that the most glaring DRY violations, like "heading" debug output logs or the code to source `testenv_constants.sh` could be moved to helpers provided by `testenv_constants.sh` itself?

**Comment by davelsan** [comment]
The models folders are now defined in the constants file.
DRYed up the code a little bit. Nothing fancy, but there was some duplications in logic and output messages that I could avoid.

**Review by vsuaste — APPROVED** [review]

**Comment by vsuaste** [comment]
If you want to do a little bit of more clean up. I think files from the patches folder: `individual_validate_joi.patch` and `individual_validate_sequelize_patch` are outdated.

**Comment by davelsan** [comment]
> If you want to do a little bit of more clean up. I think files from the patches folder: `individual_validate_joi.patch` and `individual_validate_sequelize_patch` are outdated.

Thank you for the suggestion, @vsuaste. I have removed the unused patches.

**Review by wunderbarr — APPROVED** [review]

**Review by coeit — APPROVED** [review]

**Pull request merged**